File: 2021/08/02/main.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hs = {}
        for i in range(len(nums)):
            n = nums[i]
            if n in hs:
                return [hs[n], i]
            hs[target - n] = i
        

File: 2021/08/02/main.go
package main

func twoSum(nums []int, target int) []int {
	hs := map[int]int{}
	for i := 0; i < len(nums); i++ {
		n := nums[i]
		_, exists := hs[n]
		if exists {
			return []int{hs[n], i}
		}
		hs[target-n] = i
	}
	return []int{-1, -1}
}
